src/snapred/backend/recipe/algorithm/AlignAndFocusReductionAlgorithm.py

```diff
@@ -47,9 +50,16 @@ def PyExec(self):
         )
 
         self.mantidSnapper.executeQueue()
-        DMin = reductionIngredients.reductionState.stateConfig.focusGroups[0].dMin
-        DMax = reductionIngredients.reductionState.stateConfig.focusGroups[0].dMax
-        DeltaRagged = reductionIngredients.reductionState.stateConfig.focusGroups[0].dBin
+        dMinList = []
+        dMaxList = []
+        dBinList = []
+        for i in pixelGroupingParameters:
+            dMinList.append(i.dResolution.minimum)
+            dMaxList.append(i.dResolution.maximum)
+            dBinList.append(i.dRelativeResolution)
+        DMin: float = max(dMinList)
+        DMax: float = min(dMaxList)
+        DeltaRagged: float = min(dBinList) / reductionIngredients.reductionState.instrumentConfig.NBins
```

Comment:
`DMin`, `DMax`, and `DeltaRagged` would have originally been lists.  They should probably still be lists.

Reply:
If these lists are going to be passed to a RaggedRebin later, then some protection needs to be made that the lists are in order of ascending group ID.  Or else it will create surprising and errant results.